### PR TITLE
Adding support for publishing messages with an expiration.

### PR DESCRIPTION
**Pull request by ajmckee**
- Added WithPublishOptionsExpiration which permits message expiry when the TTL has been reached
- Introduced new message var to permit the confidiontal injection of amp.Publishing options in the future.

### CODE REVIEW

**Comment by ajmckee** [comment]
I needed to add support for publishing messages with an expiration. Therefore find a basic PR for this. Additionally as more options are supported I thought it would be a good idea to add some conditional to the injection of amqp.Publishing options. I was not able to determine a method to write tests for this, but if you have suggests I would gladly write some.

File: publish.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ type PublishOptions struct {
 	ContentType string
 	// Transient or Persistent
 	DeliveryMode uint8
+	// Expiration time in ms a message will expire from a queue.
+	Expiration		string
```
Comment (wagslane) [inline publish.go:44]: Can we add a link to the description in the comment: https://www.rabbitmq.com/ttl.html#per-message-ttl-in-publishers

File: publish.go [hunk 2]
```diff
@@ -184,17 +194,30 @@ func (publisher *Publisher) Publish(
...
+		// Message Body
+		message.Body = data
+
+		// If no header options, don't add.
+		if len(options.Headers) > 0 {
+			message.Headers = tableToAMQPTable(options.Headers)
+		}
+
+		// If we have a TTL use it.
+		if options.Expiration != "" {
```
Comment (wagslane) [inline publish.go:209]: This if statement is redundant and can be removed, message.Expiration is already an empty string

File: publish.go [hunk 3]
```diff
@@ -184,17 +194,30 @@ func (publisher *Publisher) Publish(
 	}
 
 	for _, routingKey := range routingKeys {
+		var message = amqp.Publishing{}
+		message.ContentType = options.ContentType
+		message.DeliveryMode = options.DeliveryMode
+		// Message Body
```
Comment (wagslane) [inline publish.go:200]: This comment is unnecessary I think

**Comment by wagslane** [comment]
Looks like there's `go fmt` failure. Once that's resolved I'll get it merged! Thanks for making those changes

**Review by wagslane — APPROVED** [review]

**Pull request merged**